lib/pages/sport_details_page.dart

```diff
@@ -65,48 +75,64 @@ class _SportDetailsPageState extends State<SportDetailsPage> {
-                        title: Text(event['name']),
+                        title: Text(event.name ?? ''),
                         subtitle: Text(
-                          "Location: ${event['location']}\n"
-                          "Time: ${event['dateTime'].toDate()}\n"
-                          "Slots Available: ${event['slotsAvailable']}\n"
+                          "Location: ${event.location}\n"
+                          "Time: ${event.dateTime?.toDate()}\n"
+                          "Slots Available: ${event.slotsAvailable}\n"
                           "Host: $hostName",
                         ),
                         isThreeLine: true,
-                        trailing: event['slotsAvailable'] > 0
-                            ? ElevatedButton(
-                                onPressed: () async {
-                                  await firebaseService.registerForEvent(
-                                    event['id'], // Event ID
-                                    "testUserId", // Replace with logged-in user ID
-                                  );
+                        trailing: ElevatedButton(
+                          onPressed: () async {
+                            if (isRegistered) {
+                              // Unregister user
+                              await widget.firebaseService
+                                  .unregisterFromEvent(event.id!, currentUserId);
+                              ScaffoldMessenger.of(context).showSnackBar(
+                                SnackBar(
+                                    content: Text(
+                                        "Unregistered from ${event.name}")),
+                              );
+                            } else {
+                              // Register user
+                              await widget.firebaseService
+                                  .registerForEvent(event.id!, currentUserId);
+                              GoRouter.of(context).goNamed(
+                                RouteNames.registrationConfirmation,
+                                pathParameters: {'sportId': event.sportId!},
+                              );
+                            }
 
-                                  setState(() {}); // Refresh UI
-                                  ScaffoldMessenger.of(context).showSnackBar(
-                                    SnackBar(
-                                        content:
-                                            Text("Registered for ${event['name']}")),
-                                  );
-                                },
-                                child: const Text("Register"),
-                              )
-                            : const Text(
-                                "Full",
-                                style: TextStyle(color: Colors.red),
-                              ),
+                            // Refresh the UI
+                            setState(() {});
```

Comment:
I think you only have to render the UI only when the user unregisters. When the user registers, the app will route to the /confirmed animation screen so there's no need to update the page that user was previously.